src/models/blockchain/block/block_header.rs

```diff
@@ -281,6 +283,48 @@ pub(crate) mod block_header_tests {
             guesser_digest: rng.random(),
         }
     }
+    proptest::prop_compose! {
+        pub fn block_header_strategy()(
+            version_raw in any::<u64>(),
+            height_raw in any::<u64>(),
+            prev_digest_raw in proptest::collection::vec(any::<u64>(), 5),
+            timestamp_raw in any::<u64>(),
+            nonce_raw in proptest::collection::vec(any::<u64>(), 5),
+            pow_raw in proptest::array::uniform6(any::<u32>()),
+            difficulty_raw in proptest::array::uniform5(any::<u32>()),
+            guesser_digest_raw in proptest::collection::vec(any::<u64>(), 5),
```

Comment:
Both `BFieldElement` and `Digest` implement `arbitrary::Arbitrary`. You can directly get a strategy for the correct type by using the `proptest_arbitrary_interop::arb` strategy.